src/physics/negative_spalart_allmaras_rans_model.cpp

```diff
@@ -0,0 +1,718 @@
+    }
+    else{
+        std::cout << "ERROR in physics/reynolds_averaged_navier_stokes.cpp --> compute_coefficient_f_v1(): real2 != real or FadType" << std::endl;
+        std::abort();
+    }
+
+    return coefficient_f_v1;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+real ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_f_v2 (
+    const real &coefficient_Chi) const
+{
+    // Compute coefficient f_v2
+    real coefficient_f_v2;
+    const real coefficient_f_v1 = compute_coefficient_f_v1(coefficient_Chi);
+
+    coefficient_f_v2 = 1.0-coefficient_Chi/(1.0+coefficient_Chi*coefficient_f_v1);
+
+    return coefficient_f_v2;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template<typename real2>
+real2 ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_f_n (
+    const real2 &nu_tilde,
+    const real2 &laminar_kinematic_viscosity) const
+{
+    // Compute coefficient f_n
+    real2 coefficient_f_n;
+    const real2 coefficient_Chi = compute_coefficient_Chi(nu_tilde,laminar_kinematic_viscosity);
+
+    if constexpr(std::is_same<real2,real>::value){ 
+        if (nu_tilde>=0.0)
+            coefficient_f_n = 1.0;
+        else
+            coefficient_f_n = (c_n1+coefficient_Chi*coefficient_Chi*coefficient_Chi)/(c_n1-coefficient_Chi*coefficient_Chi*coefficient_Chi);
+    }
+    else if constexpr(std::is_same<real2,FadType>::value){
+        const FadType const_one_fad = 1.0; 
+        if (nu_tilde>=0.0)
+            coefficient_f_n = const_one_fad;
+        else
+            coefficient_f_n = (c_n1_fad+coefficient_Chi*coefficient_Chi*coefficient_Chi)/(c_n1_fad-coefficient_Chi*coefficient_Chi*coefficient_Chi);
+    }
+    else{
+        std::cout << "ERROR in physics/reynolds_averaged_navier_stokes.cpp --> compute_coefficient_f_n(): real2 != real or FadType" << std::endl;
+        std::abort();
+    }
+
+    return coefficient_f_n;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+real ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_f_t2 (
+    const real &coefficient_Chi) const
+{
+    // Compute coefficient f_t2
+    real coefficient_f_t2;
+
+    coefficient_f_t2 = c_t3*exp(-c_t4*coefficient_Chi*coefficient_Chi);
+
+    return coefficient_f_t2;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+real ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_f_w (
+    const real &coefficient_g) const
+{
+    // Compute coefficient f_w
+    real coefficient_f_w;
+
+    coefficient_f_w = coefficient_g*pow((1.0+pow(c_w3,6.0))/(pow(coefficient_g,6.0)+pow(c_w3,6.0)),1.0/6.0);
```

Comment:
`pow(c_w3,6.0)` is computed twice; pre-compute and store in a variable for efficiency

Reply:
fixed